backend/src/controllers/RestaurantController.ts

```diff
@@ -62,6 +62,21 @@ const searchRestaurants = async (req: Request, res: Response) => {
   }
 };
 
+const getRestaurant = async (req: Request, res: Response) => {
+  try {
+    const restaurantId = req.params.restaurantId;
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/45#issuecomment-2411028139>#43cd30</a><span> - 10/14/2024, 12:06 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The getRestaurant function should validate and sanitize the restaurantId parameter to prevent potential injection attacks or invalid ObjectId errors. A similar issue was also found in backend/src/controllers/RestaurantController.ts (line 67).
</div>
<div id="fix">
<b>Fix:</b> Implement input validation for the restaurantId parameter. Use a library like mongoose-objectid-validator or a custom function to ensure the restaurantId is a valid ObjectId before using it in the database query.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -65,7 +65,13 @@
  const getRestaurant = async (req: Request, res: Response) => {
    try {
      const restaurantId = req.params.restaurantId;
 +    if (!mongoose.Types.ObjectId.isValid(restaurantId)) {
 +      return res.status(400).json({ message: 'Invalid restaurant ID' });
 +    }
 +
      const restaurant = await Restaurant.findById(restaurantId).lean();
 +
      if (!restaurant) {
        return res.status(404).json({ message: 'Restaurant not found' });
      }
 +    res.json(restaurant);
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged